src/lib_json/json_writer.cpp

```diff
@@ -141,14 +142,15 @@ std::string valueToString(double value, bool useSpecialFloats, unsigned int prec
   char buffer[32];
   int len = -1;
 
-  char formatString[6];
-  sprintf(formatString, "%%.%dg", precision);
+  std::stringstream value_ss;
+  value_ss << std::setprecision( (int)precision ) << value;
+  std::string value_str = value_ss.str();
 
   // Print into the buffer. We need not request the alternative representation
   // that always has a decimal point because JSON doesn't distingish the
   // concepts of reals and integers.
   if (isfinite(value)) {
-    len = snprintf(buffer, sizeof(buffer), formatString, value);
+    len = snprintf(buffer, sizeof(buffer), (const char*)value_str.c_str());
```

Comment:
Same as other issue, clang complaints about this.
My only regret is about memory usage of stringstream -> string -> c_string.
What's your opinion?

2016-02-12 16:56 GMT+01:00 Billy Donahue <notifications@github.com>:

> In src/lib_json/json_writer.cpp
> <https://github.com/open-source-parsers/jsoncpp/pull/421#discussion_r52758334>
> :
>
> >
> >    // Print into the buffer. We need not request the alternative representation
> >    // that always has a decimal point because JSON doesn't distingish the
> >    // concepts of reals and integers.
> >    if (isfinite(value)) {
> > -    len = snprintf(buffer, sizeof(buffer), formatString, value);
> > +    len = snprintf(buffer, sizeof(buffer), (const char*)value_str.c_str());
>
> std::string::c_str already returns const char*. Why cast this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/open-source-parsers/jsoncpp/pull/421/files#r52758334>.
>



-- 
Fiorentino Ing. Stefano
Cell. +39 340 2216385